WORKSPACE

```diff
@@ -205,3 +207,30 @@ http_archive(
     strip_prefix = "abseil-py-1.2.0",
     urls = ["https://github.com/abseil/abseil-py/archive/refs/tags/v1.2.0.tar.gz"],
 )
+
+# zlib is imported through protobuf. Make the dependency explicit considering
+# it's used outside protobuf.
+maybe(
+    http_archive,
+    name = "zlib",
+    build_file = "@com_google_protobuf//:third_party/zlib.BUILD",
+    sha256 = "91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9",
+    strip_prefix = "zlib-1.2.12",
+    urls = [
+        "https://zlib.net/zlib-1.2.12.tar.gz",
+        "https://zlib.net/fossils/zlib-1.2.12.tar.gz",
+    ],
+)
+
+git_repository(
+    name = "boringssl",
+    # Commits must come from the main-with-bazel branch.
+    commit = "b9232f9e27e5668bc0414879dcdedb2a59ea75f2",
+    # Use the GitHub mirror because the official source at
+    # https://boringssl.googlesource.com/boringssl does not allow
+    # unauthenticated git clone and the archives suffer from
+    # https://github.com/google/gitiles/issues/84 preventing the use of
+    # sha256sum on archives.
```

Comment:
Is this really an issue here ? The generated github zip-archives are guaranteed to generate stable files, so we can rely on them being sha256-able. With that in mind, I prefer if we do it the same way like the other external references in this `WORKSPACE` file.

Also we tend to name the libraries with a more globally unique namespaced form, so instead of `"boringssl"` maybe better `"com_google_boringssl"` ?
```bzl
# For sha256
http_archive(
    name = "com_google_boringssl",
    sha256 = "7128f303ded3a647c4ff08d9396ac86a03f3f2f8ab3080fe8183f9092860736d",
    strip_prefix = "boringssl-b9232f9e27e5668bc0414879dcdedb2a59ea75f2",
    urls = ["https://github.com/google/boringssl/archive/b9232f9e27e5668bc0414879dcdedb2a59ea75f2.zip"],
)
```

Reply:
Done. Thanks! I didn't know Github provides that.